spec/API_specification/indexing.md

```diff
@@ -134,7 +134,11 @@ Multi-dimensional arrays must extend the concept of single-axis indexing to mult
 -   Each axis may be independently indexed via single-axis indexing by providing a comma-separated sequence ("selection tuple") of single-axis indexing expressions (e.g., `A[:, 2:10, :, 5]`).
 
     ```{note}
-    In Python, `x[(exp1, exp2, ..., expN)]` is equivalent to `x[exp1, exp2, ..., expN]`; the latter is syntactic sugar for the former.
+    In Python, `A[(exp1, exp2, ..., expN)]` is equivalent to `A[exp1, exp2, ..., expN]`; the latter is syntactic sugar for the former.
+
+    Accordingly, if `A` has rank `1`, then `A[(2:10,)]` must be equivalent to `A[2:10]`. If `A` has rank `2`, then `A[(2:10, :)]` must be equivalent to `A[2:10, :]`. And so on and so forth.
+
+    To access the value in a zero-dimensional array, one must provide an empty tuple; e.g., `A[()]`.
```

Comment:
I had just assumed the Array API didn't support 0d indexing, so just incase I'll ask—are we sure about allowing empty tuples to index 0d arrays?

Reply:
This does work, because the empty tuple indexes zero axes. 

However, I don't like this sentence "To access the value in a zero-dimensional array" now that I look at it. In the spec, indexing returns an array, so `a[()]` would just return `a` unchanged (regardless of its shape). "Access the value" seems to imply some sort of NumPy-like scalar objects.

Reply:
Right. Okay. The preferred means of accessing the value in a 0D array would be, e.g., `float(x)` and `int(x)`, correct?

If so, I'll remove that sentence, and add a sentence stating that indexing zero-dimensional arrays is not allowed.

Reply:
> The preferred means of accessing the value in a 0D array would be, e.g., float(x) and int(x), correct?

If you want a Python scalar yes. The 0-D array itself ought to just work as a scalar value in most relevant contexts inside of the array API. 

> If so, I'll remove that sentence, and add a sentence stating that indexing zero-dimensional arrays is not allowed.

`A[()]` and `A[...]` should be allowed on 0-dimensional arrays, because in both cases the indices index 0 axes.